x/eibc/keeper/msg_server.go

```diff
@@ -254,24 +235,47 @@ func (m msgServer) UpdateDemandOrder(goCtx context.Context, msg *types.MsgUpdate
 	return &types.MsgUpdateDemandOrderResponse{}, nil
 }
 
-func (m msgServer) GetOutstandingOrder(ctx sdk.Context, orderId string) (*types.DemandOrder, error) {
-	// Check that the order exists in status PENDING
-	demandOrder, err := m.GetDemandOrder(ctx, commontypes.Status_PENDING, orderId)
+func (m msgServer) FindFulfiller(goCtx context.Context, msg *types.MsgFindFulfiller) (*types.MsgFindFulfillerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	err := msg.ValidateBasic()
 	if err != nil {
-		return nil, err
+		return nil, errorsmod.Wrap(err, "vbasic")
+	}
+
+	return &types.MsgFindFulfillerResponse{}, m.Keeper.FulfillByOnDemandLP(ctx, msg.OrderId, msg.Rng)
```

Comment:
not very typical for a Find method to not return anything. Maybe call it `MatchFulfiller` ?

Reply:
yeah true
ok will change